TestProjects/Navigation/RoutingExample.Core/ViewModels/TestCViewModel.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using MvvmCross.Core.Navigation;
+using MvvmCross.Core.ViewModels;
+using MvvmCross.Platform;
+using RoutingExample.Core.ViewModels;
+
+namespace RoutingExample.Core.ViewModels
+{
+    public class TestCViewModel
+        : MvxViewModel<User, User>
+    {
+        private readonly IMvxNavigationService _navigationService;
+        public TestCViewModel(IMvxNavigationService navigationService)
+        {
+            _navigationService = navigationService;
+        }
+
+        private string _parameter;
+        public string Parameter
+        {
+            get { return _parameter; }
+            set { SetProperty(ref _parameter, value); }
+        }
+
+        public void Init(string viewmodelcparameter)
```

Comment:
So do you mean to I should try to send the parameters through to the Prepare method?

I was just following [this](https://www.mvvmcross.com/documentation/fundamentals/navigation#uri-navigation) but I can try to change it to use the Prepare method.